layer/CMakeLists.txt

```diff
@@ -45,11 +45,10 @@ add_definitions(-DVK_ENABLE_BETA_EXTENSIONS)
 set(TARGET_NAME khronos_profiles)
 
 if (WIN32)
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS /bigobj")
-    set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS /bigobj")
+    add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+    add_compile_options(/bigobj)
 else()
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wpointer-arith")
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wpointer-arith")
```

Comment:
2 things.

1.) CMAKE_C_FLAGS isn't needed at all anymore.
2.) Unless you HAVE to don't touch these variables.